### DISCUSSION TITLE
How to access table query or records?

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#1439] [Help]
In my ListProducts component I have table generated by Filament. What I want to achieve is to add above it statistics connected to data displayed in table. To do this I have to access somehow current table query or its records.

I can easily do this by Livewire hook **updated()**, and there use **$this->getTableRecords()**. The problem is that I want to display statistics also when page is loaded, not only after **updated()** hook is invoked.

To do this I tried to use many different hooks, eg: **mount, afterFill, bootedInteractsWithTable, updatedRecords**, but some of them doesnt exists and on some table data is not yet populated, so I dont know how to access it when table is ready.

**Comment by danharrin** [comment]
Hm, I would suggest doing this logic inside a Livewire computed property (https://laravel-livewire.com/docs/2.x/properties#computed-properties) which you can access from your view. This will refresh every time the data is updated.

**Reply by studiowizjo** [reply]
I did it this way:

```php
public $statistics = [];

public function getStatisticsProperty() {
    $productsCount = 0;
    $priceTotal = 0;

    if ($this->getTableRecords()) {
        foreach ($this->getTableRecords() as $record) {
            $productsCount += $record['quantity'];
            $priceTotal += $record['quantity'] * $record->product->price;
        }
    }

    return [
        'products_count' => [
            'title' => 'Liczba produktów',
            'number' => $productsCount,
        ],
        'price_total' => [
            'title' => 'Wartość produktów',
            'number' => $priceTotal,
        ],
    ];
}
```

but it doesn`t work. I tried also using **getTableProperty()**.

So, how to hook to table records change?

**Reply by danharrin** [reply]
I wouldn't suggest using `$this->getTableRecords()`, you can access the actual Eloquent query to perform these calculations in a computed prop using `$this->getFilteredTableQuery()`.

**Reply by studiowizjo** [reply]
Thank you, really appreciate help from you.

However the problem is not about calculations, but how to invoke them just after table data has been loaded. There is not problem with that later, when user do something with table (sort, search or filter) because I use Liverwire update hook to call my calculateStatistics() method, but I cannot find any way to do this on page load.

I checked all Livewire lifecycle hooks (https://laravel-livewire.com/docs/2.x/lifecycle-hooks) as well as Filament (https://filamentphp.com/docs/2.x/admin/resources#lifecycle-hooks), but none of them solves this on ListRecords. Filament hooks seems to be firing only on EditRecord and CreateRecord pages and about Livewire hooks:
- booted - table data is not avalaible yet
- updatedTable - not sure why this is not firing, as there is such property on ListRecords page
- also any suggested above computed property I checked didnt worked

I couldnt find also anything usefull for that digging in package or Livewire source code.
